packages/validator/src/lib.js

```diff
@@ -464,87 +502,94 @@ const validate = async (delegation, config) => {
+const verifyAuthorization = async (delegation, proofs, config) => {
+  const issuer = delegation.issuer.did()
+  // If the issuer is a did:key we just verify a signature
+  if (issuer.startsWith('did:key:')) {
+    return verifySignature(delegation, config.principal.parse(issuer))
+  }
+  // If the issuer is the root authority we use authority itself to verify
+  else if (issuer === config.authority.did()) {
+    return verifySignature(delegation, config.authority)
+  } else {
+    // If issuer is not a did:key principal nor configured authority, we
+    // attempt to resolve embedded authorization session from the authority.
+    const session = await verifySession(delegation, proofs, config)
+    // If we have valid session we consider authorization valid
+    if (!session.error) {
+      return delegation
+    } else if (session.failedProofs.length > 0) {
+      return new SessionEscalation({ delegation, cause: session })
+    }
+    // Otherwise we try to resolve did:key from the DID instead
+    // and use that to verify the signature
+    else {
+      const verifier = await config.resolveDIDKey(issuer)
+      if (verifier.error) {
+        return verifier
+      } else {
+        return verifySignature(
+          delegation,
+          config.principal.parse(verifier).withDID(issuer)
+        )
+      }
+    }
   }
-
-  const valid = await UCAN.verifySignature(delegation.data, verifier)
-  return valid ? delegation : new InvalidSignature(delegation, verifier)
 }
 
 /**
- * @param {API.DID} did
- * @param {API.Delegation} delegation
- * @param {Required<API.ClaimOptions>} config
- * @returns {Promise<API.Result<API.Verifier, API.DIDKeyResolutionError>>}
+ * @template {API.Delegation} T
+ * @param {T} delegation
+ * @param {API.Verifier} verifier
+ * @returns {Promise<API.Result<T, API.InvalidSignature|API.DIDKeyResolutionError>>}
  */
-const resolveVerifier = async (did, delegation, config) => {
-  if (did === config.authority.did()) {
-    return config.authority
-  } else if (did.startsWith('did:key:')) {
-    return config.principal.parse(did)
-  } else {
-    // First we attempt to resolve key from the embedded proofs
-    const local = await resolveDIDFromProofs(did, delegation, config)
-    const result = !local?.error
-      ? local
-      : // If failed to resolve because there is an invalid proof propagate error
-      (local?.cause?.failedProofs?.length || 0) > 0
-      ? local
-      : // otherwise either use resolved key or if not found attempt to resolve
-        // did externally
-        await config.resolveDIDKey(did)
-    return result.error ? result : config.principal.parse(result).withDID(did)
-  }
+const verifySignature = async (delegation, verifier) => {
+  const valid = await UCAN.verifySignature(delegation.data, verifier)
+  return valid ? delegation : new InvalidSignature(delegation, verifier)
 }
 
 /**
- * @param {API.DID} did
+ * Attempts to find an authorization session - an `./update` capability
```

Comment:
does this `./update` still make sense?

can we link to which spec defines `ucan/attest` behavior?